In [20]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

### AskWomen Data Cleaning and EDA

In [21]:
women_df = pd.read_csv('./Data/women_posts.csv')

In [22]:
women_df.head()

all_awardings  allow_live_comments             author  \
0            []                False           C4TERW4L   
1            []                False   throwawayalientk   
2            []                False   throwawayalientk   
3            []                False    Exact_Roll_4048   
4            []                False  Unfair_Salad_5521   

  author_flair_css_class author_flair_richtext author_flair_text  \
0                    NaN                    []               NaN   
1                    NaN                    []               NaN   
2                    NaN                    []               NaN   
3                    NaN                    []               NaN   
4                    NaN                    []               NaN   

  author_flair_type author_fullname  author_is_blocked author_patreon_flair  \
0              text        t2_w635x              False                False   
1              text     t2_5992f1cb              False                False   
2              text     t2_5992f1cb              False                False   
3              text     t2_7dtxul8b              False                False   
4              text     t2_ha34dlxa              False                False   

   ... author_flair_background_color author_flair_text_color  banned_by  \
0  ...                           NaN                     NaN        NaN   
1  ...                           NaN                     NaN        NaN   
2  ...                           NaN                     NaN        NaN   
3  ...                           NaN                     NaN        NaN   
4  ...                           NaN                     NaN        NaN   

   call_to_action  category author_flair_template_id link_flair_template_id  \
0             NaN       NaN                      NaN                    NaN   
1             NaN       NaN                      NaN                    NaN   
2             NaN       NaN                      NaN                    NaN   
3             NaN       NaN                      NaN                    NaN   
4             NaN       NaN                      NaN                    NaN   

  link_flair_text author_cakeday  edited  
0             NaN            NaN     NaN  
1             NaN            NaN     NaN  
2             NaN            NaN     NaN  
3             NaN            NaN     NaN  
4             NaN            NaN     NaN  

[5 rows x 73 columns]

In [23]:
women_df.shape

(3000, 73)

In [24]:
women_df['title'].isnull().sum(), women_df['selftext'].isnull().sum()

(0, 923)

- There are no nulls in the 'title' column and 923 nulls in the 'selftext' column.